packages/core-sdk/resolvers/token-resolvers/resolvers/1inch/src/query/functions/getTokenComponents.ts

```diff
@@ -0,0 +1,95 @@
+import { BigInt } from "@web3api/wasm-as";
+import { Big } from "as-big";
+
+import {
+  env,
+  Ethereum_Connection,
+  Ethereum_Query,
+  Input_getTokenComponents,
+  Interface_Token,
+  Interface_TokenComponent,
+  QueryEnv,
+  Token_Query,
+  Token_TokenType,
+} from "../w3";
+
+function getPoolTokenAddresses(poolAddress: string, connection: Ethereum_Connection): string[] {
+  const tokenAddressRes = Ethereum_Query.callContractView({
+    address: poolAddress,
+    method: "function getTokens() external view returns(address[] tokens)",
+    args: [],
+    connection: connection,
+  });
+  if (tokenAddressRes.isErr) {
+    throw new Error("Invalid protocol token");
+  }
+  return tokenAddressRes.unwrap().split(",");
+}
+
+export function getTokenComponents(input: Input_getTokenComponents): Interface_TokenComponent {
+  if (env == null) throw new Error("env is not set");
+  const connection = (env as QueryEnv).connection;
+
+  const token = Token_Query.getToken({
+    address: input.tokenAddress,
+    m_type: Token_TokenType.ERC20,
+  }).unwrap();
+
+  if (!token) {
+    throw new Error(`Token ${input.tokenAddress} is not a valid ERC20 token`);
+  }
+
+  const poolTokenAddresses: string[] = getPoolTokenAddresses(token.address, connection);
+
+  const tokenDecimals: string = BigInt.fromUInt16(10).pow(token.decimals).toString();
+  const totalSupply: Big = Big.of(token.totalSupply.toString()) / Big.of(tokenDecimals);
+
+  const components: Interface_TokenComponent[] = [];
+  let unresolvedComponents: i32 = 0;
+
+  for (let j = 0; j < poolTokenAddresses.length; j++) {
+    // get underlying token
+    const underlyingTokenAddress: string = poolTokenAddresses[j];
+    const underlyingToken: Interface_Token = changetype<Interface_Token>(
+      Token_Query.getToken({
```

Comment:
Need to handle case for native token: 0x0 in getToken function

Reply:
Done

Reply:
Done